### PR TITLE
[clang][powerpc][wasm][systemz][x86] Replace target vector popcount intrinsics with __builtin_elementwise_popcount

### PR DESCRIPTION
**Pull request by RKSimon**
Now that we have the C/C++ `__builtin_elementwise_popcount` intrinsic (#108121) - remove custom target intrinsics that just immediately map to Intrinsic::ctpop and use the generic intrinsic directly.

### CODE REVIEW

**Review by tlively — COMMENTED** [review]
WebAssembly changes LGTM.

File: clang/lib/Headers/avx512bitalgintrin.h [hunk 1]
```diff
@@ -23,7 +23,7 @@
 static __inline__ __m512i __DEFAULT_FN_ATTRS
 _mm512_popcnt_epi16(__m512i __A)
 {
-  return (__m512i) __builtin_ia32_vpopcntw_512((__v32hi) __A);
+  return (__m512i)__builtin_elementwise_popcount((__v32hi)__A);
```
Comment (phoebewang) [inline clang/lib/Headers/avx512bitalgintrin.h:26]: hi or hu? The same below.

Comment (RKSimon) [reply clang/lib/Headers/avx512bitalgintrin.h]: I've updated all the x86 cases to use unsigned for consistency - `__builtin_elementwise_popcount` doesn't really care, although it expects src/dst types to match (although x86 casts back to _mXXXi types anyway).

**Review by phoebewang — APPROVED** [review]
LGTM.

**Comment by uweigand** [comment]
SystemZ changes LGTM.

**Comment by RKSimon** [comment]
@nemanjai Are you OK with the PPC changes?

**Comment by chenzheng1030** [comment]
> @nemanjai Are you OK with the PPC changes?

on behalf of @nemanjai : ): the change LG to PPC.

FYI @lei137 @stefanp-ibm if we doc these builtins for PPC users, we may need to change the the docs.

**Comment by RKSimon** [comment]
Officially we don't guarantee target intrinsics at all - but I can add a release note if it will help?

**Comment by chenzheng1030** [comment]
> Officially we don't guarantee target intrinsics at all - but I can add a release note if it will help?

hmm, I meant IBM OpenXLC compiler product doc, many builtins are listed there, not the community clang/llvm docs.

**Pull request merged**